### ISSUE TITLE
Hasn't been working recently

### ISSUE THREAD

**Issue by SavannahJohnston** [jscher2000/De-Troll-YouTube---Hide-Unwanted-Commenters#7]
In the past few days the trash can symbol is never showing up, and using the context menu isn't working either; it always says I didn't click inside the comment, even when I did (and even though clicking just the name always worked before). I'm used to the trash can symbol _occasionally_ not showing up, but usually a refresh fixes it, and for the past several days it just refuses to show up. I guess it could be one of my other add-ons interfering, but I've had this setup for ages and De-Troll has worked fine the whole time until recently. Did YouTube recently have an update that might have broken De-Troll? Or should I start experimenting with turning various add-ons off? (I've already tried all the common-sense ones, like my adblocker and all the ones that explicitly affect YouTube, e.g. Unhook.)

**Comment by jscher2000** [comment]
Thank you for mentioning this. YouTube changed some of their tags so I had to make a few corresponding changes. Hopefully version 0.8.1 will work for you when it goes live on the site later tonight or tomorrow.

**Comment by SavannahJohnston** [comment]
Thanks! I've downloaded the update, and what's happening now is that it works for top-level comments but not replies. The trash can symbol doesn't show up next to replies, and when I try the context menu I don't get an error message but also nothing else happens.

I also am no longer getting the preferences tab on the add-on configuration page, so I seem to have no way to remove users from the blocklist anymore.

Disabling the YouTube-related add-ons and restarting my browser just in case…

Ah, interesting. The preferences tab is showing up now, so I guess the browser just needed a restart for that. Upon examination, the users I've blocked via context menu do appear on the list, but their comments are still showing up just the same. Only a user that I blocked with the trash can button is actually hidden from view. Not sure if that's just because the button is somehow doing something different from what the context menu does (seems doubtful) or if the add-on is just treating top-level comments differently.

**Comment by jscher2000** [comment]
I missed that. Version 0.8.2 should fix the comment replies.

> I also am no longer getting the preferences tab on the add-on configuration page, so I seem to have no way to remove users from the blocklist anymore.

I think this happens if the Add-ons page is open during an update. You can try a hard reload (Ctrl+Shift+R, or on Mac, Command+Shift+R) to work around it next time.

**Comment by SavannahJohnston** [comment]
Seems to be working perfectly now! Thanks so much!

> I think this happens if the Add-ons page is open during an update. You can try a hard reload (Ctrl+Shift+R, or on Mac, Command+Shift+R) to work around it next time.

That works! Thanks for the tip!

**Comment by jscher2000** [comment]
Thank you for testing and confirming.

**Comment by artcytracy** [comment]
> Thank you for mentioning this. YouTube changed some of their tags so I had to make a few corresponding changes. Hopefully version 0.8.1 will work for you when it goes live on the site later tonight or tomorrow.

Can I trouble you to ask what tags specifically? 

I am looking at an issue in another similar extension that appears to be impacted by these changes.

**Comment by SavannahJohnston** [comment]
I think this one has stopped working again…

**Comment by jscher2000** [comment]
No trash can, no blocking, or both?

I'm not noticing a problem yet. Google is notorious for testing different page layouts on groups of users, so you might be seeing a temporary change or maybe you're seeing what everyone will see soon. 

Maybe what you could do is right-click the comment author's name, then click Inspect. This will open the Inspector tool with the author name selected. Right-click that and expand Copy and click CSS Path. Here's the one I get:

html body ytd-app div#content.style-scope.ytd-app ytd-page-manager#page-manager.style-scope.ytd-app ytd-watch-flexy.style-scope.ytd-page-manager.watch-root-element.hide-skeleton div#columns.style-scope.ytd-watch-flexy div#primary.style-scope.ytd-watch-flexy div#primary-inner.style-scope.ytd-watch-flexy div#below.style-scope.ytd-watch-flexy ytd-comments#comments.style-scope.ytd-watch-flexy ytd-item-section-renderer#sections.style-scope.ytd-comments div#contents.style-scope.ytd-item-section-renderer ytd-comment-thread-renderer.style-scope.ytd-item-section-renderer ytd-comment-view-model#comment.style-scope.ytd-comment-thread-renderer div#body.style-scope.ytd-comment-view-model div#main.style-scope.ytd-comment-view-model div#header.style-scope.ytd-comment-view-model div#header-author.style-scope.ytd-comment-view-model h3.style-scope.ytd-comment-view-model a#author-text.yt-simple-endpoint.style-scope.ytd-comment-view-model span.style-scope.ytd-comment-view-model

If yours is different, that might allow me to start making changes now without waiting.

**Comment by jscher2000** [comment]
> > Thank you for mentioning this. YouTube changed some of their tags so I had to make a few corresponding changes. Hopefully version 0.8.1 will work for you when it goes live on the site later tonight or tomorrow.
> 
> Can I trouble you to ask what tags specifically?
> 
> I am looking at an issue in another similar extension that appears to be impacted by these changes.

Whoops, sorry, I missed this. You could use the code comparison page (this is linked from the comment next to the content script): https://github.com/jscher2000/De-Troll-YouTube---Hide-Unwanted-Commenters/commit/fa798f9a68cbac519480ba800eb81e6102f9144c 

Actually, the changes were split over two fixes. This was the earlier one: https://github.com/jscher2000/De-Troll-YouTube---Hide-Unwanted-Commenters/commit/f825c816995efb562eab34a475ca299f4665dcb9

**Comment by SavannahJohnston** [comment]
> No trash can, no blocking, or both?

Sorry for the lack of detail before!

The trash cans weren't showing up. They're showing up again on video pages, which is weird because I haven't restarted my browser or anything. (In fact I had just done a full computer restart when I noticed they weren't showing up, which is why I thought it was definitely the extension not working, as opposed to just an issue caused by my browser being open for too long or something.) But, for the past week or so the trash can hasn't been showing up on channel pages again. I've restarted my browser several times today for unrelated reasons, and restarted my whole computer for another unrelated reason, so I don't think it's my browser acting up, but of course I can't know for sure.

If I inspect a comment author's name on a video page and copy CSS Path, what I get is:

html.onhwzxys.wxgdmmh body ytd-app div#content.style-scope.ytd-app ytd-page-manager#page-manager.style-scope.ytd-app ytd-watch-flexy.style-scope.ytd-page-manager.watch-root-element.hide-skeleton div#columns.style-scope.ytd-watch-flexy div#primary.style-scope.ytd-watch-flexy div#primary-inner.style-scope.ytd-watch-flexy div#below.style-scope.ytd-watch-flexy ytd-comments#comments.style-scope.ytd-watch-flexy ytd-item-section-renderer#sections.style-scope.ytd-comments div#contents.style-scope.ytd-item-section-renderer ytd-comment-thread-renderer.style-scope.ytd-item-section-renderer ytd-comment-view-model#comment.style-scope.ytd-comment-thread-renderer div#body.style-scope.ytd-comment-view-model div#main.style-scope.ytd-comment-view-model div#header.style-scope.ytd-comment-view-model div#header-author.style-scope.ytd-comment-view-model h3.style-scope.ytd-comment-view-model a#author-text.yt-simple-endpoint.style-scope.ytd-comment-view-model span.style-scope.ytd-comment-view-model

If I do the same on the comment author's channel page (selecting their name at the top), what I get is:

html.onhwzxys.wxgdmmh body ytd-app div#content.style-scope.ytd-app ytd-page-manager#page-manager.style-scope.ytd-app ytd-browse.style-scope.ytd-page-manager div#header.style-scope.ytd-browse ytd-tabbed-page-header.style-scope.ytd-browse.grid.grid-5-columns tp-yt-app-header-layout.style-scope.ytd-tabbed-page-header div#wrapper.style-scope.tp-yt-app-header-layout tp-yt-app-header#header.style-scope.ytd-tabbed-page-header div#contentContainer.style-scope.tp-yt-app-header div#page-header-container.style-scope.ytd-tabbed-page-header div#page-header.style-scope.ytd-tabbed-page-header yt-page-header-renderer.page-header-renderer-wiz.style-scope.ytd-tabbed-page-header yt-page-header-view-model.page-header-view-model-wiz.page-header-view-model-wiz--no-padding div.page-header-view-model-wiz__page-header-content div.page-header-view-model-wiz__page-header-headline div.page-header-view-model-wiz__page-header-headline-info yt-dynamic-text-view-model.dynamic-text-view-model-wiz.page-header-view-model-wiz__page-header-title.page-header-view-model-wiz__page-header-title--page-header-title-large h1.dynamic-text-view-model-wiz__h1 span.yt-core-attributed-string.yt-core-attributed-string--white-space-pre-wrap

If I select their @ name instead of their channel name (next to where it says the number of subscribers), what I get is:

html.onhwzxys.wxgdmmh body ytd-app div#content.style-scope.ytd-app ytd-page-manager#page-manager.style-scope.ytd-app ytd-browse.style-scope.ytd-page-manager div#header.style-scope.ytd-browse ytd-tabbed-page-header.style-scope.ytd-browse.grid.grid-5-columns tp-yt-app-header-layout.style-scope.ytd-tabbed-page-header div#wrapper.style-scope.tp-yt-app-header-layout tp-yt-app-header#header.style-scope.ytd-tabbed-page-header div#contentContainer.style-scope.tp-yt-app-header div#page-header-container.style-scope.ytd-tabbed-page-header div#page-header.style-scope.ytd-tabbed-page-header yt-page-header-renderer.page-header-renderer-wiz.style-scope.ytd-tabbed-page-header yt-page-header-view-model.page-header-view-model-wiz.page-header-view-model-wiz--no-padding div.page-header-view-model-wiz__page-header-content div.page-header-view-model-wiz__page-header-headline div.page-header-view-model-wiz__page-header-headline-info yt-content-metadata-view-model.page-header-view-model-wiz__page-header-content-metadata.yt-content-metadata-view-model-wiz.yt-content-metadata-view-model-wiz--inline.yt-content-metadata-view-model-wiz--medium-text div.yt-content-metadata-view-model-wiz__metadata-row.yt-content-metadata-view-model-wiz__metadata-row--metadata-row-inline span.yt-core-attributed-string.yt-content-metadata-view-model-wiz__metadata-text.yt-core-attributed-string--white-space-pre-wrap.yt-core-attributed-string--link-inherit-color
